src/structures/Message.js

```diff
@@ -406,15 +406,15 @@ class Message extends Base {
   }
 
   /**
-   * The Message this crosspost/reply/pin-add references, if cached
-   * @type {?Message}
-   * @readonly
+   * Fetches the Message this crosspost/reply/pin-add references, if available to the client
+   * @returns {Promise<Message>}
    */
-  get referencedMessage() {
-    if (!this.reference) return null;
-    const referenceChannel = this.client.channels.resolve(this.reference.channelID);
-    if (!referenceChannel) return null;
-    return referenceChannel.messages.resolve(this.reference.messageID);
+  fetchReference() {
```

Comment:
This function needs to be marked as async (or use `return Promise.reject()`) so errors can be handled with `.catch()` handlers.
```suggestion
  async fetchReference() {
```

Reply:
Should this also await on the last line? Linter will complain about an unused async otherwise.

Reply:
linter will complain about a useless await if you add it after a return. I think returen Promise.reject is a better option

Reply:
I'm pretty sure the rule checks for the `await` and `throw` keywords, as `async` changes their behaviour.